src/test/java/org/jenkinsci/plugins/workflow/steps/EnvStepTest.java

```diff
@@ -245,40 +158,48 @@
         });
     }
 
-    @Test public void mapNumbersNested() {
+    @Test public void mapArguments() {
         story.addStep(new Statement() {
             @Override public void evaluate() throws Throwable {
                 WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
                 p.setDefinition(new CpsFlowDefinition(
                     "node {\n" +
-                        "  withEnv([A: 1]) {\n" +
-                        "    withEnv([B: 2]) {\n" +
-                        "      withEnv([C: true]) {\n" +
-                        "        isUnix() ? sh('echo A=$A B=$B C=$C') : bat('echo A=%A% B=%B% C=%C%')\n" +
-                        "      }\n" +
-                        "    }\n" +
+                        "  withEnv(a: 1, b: 2, c: 'hello world', d: true) {\n" +
+                        "    echo \"a=$a b=$b c=$c d=$d\"" +
```

Comment:
arguably more legibly

```suggestion
                        "    echo(/a=$a b=$b c=$c d=$d/)" +
```